built_json/lib/src/built_list_serializer.dart

```diff
@@ -11,27 +11,43 @@ class BuiltListSerializer implements Serializer<BuiltList> {
   final String wireName = 'list';
 
   @override
-  Object serialize(Serializers serializers, BuiltList object,
-      {GenericType genericType: const GenericType()}) {
-    final valueGenericType = genericType.parameters.isEmpty
-        ? const GenericType()
-        : genericType.parameters[0];
-
-    return object.map(
-        (item) => serializers.serialize(item, genericType: valueGenericType));
+  Object serialize(Serializers serializers, BuiltList builtList,
+      {FullType specifiedType: const FullType()}) {
+    final isUnderspecified =
+        specifiedType.isObject || specifiedType.parameters.isEmpty;
```

Comment:
> What if it's not Object but also not BuiltList?

This should be checked for statically, i.e. because we generate calls to this code. It's not yet clear if we need to also check dynamically.